replace st and eval command line handlers by clap commands

demarey opened the pull request.

demarey committed:
Add st and eval Clap commands

demarey committed:
move class var used for cli tests in ClapPharoCommandsTest

demarey committed:
typos

demarey committed:
update cli tests

demarey committed:
fix st and eval commands behaviour

demarey committed:
Delete old eval and st command line handlers

jecisc commented:
```
+ /builds/workspace/est_and_branch_Pipeline_PR-18514/64/build/bootstrap-downloads/vmtarget/pharo --headless Pharo14.0-PR-compiler-64bit-30a76a493a.image --no-default-preferences st /builds/workspace/est_and_branch_Pipeline_PR-18514/64/bootstrap/scripts/01-initialization/01-init.st --no-source --save --quit

Error: Unrecognized arguments: --no-source, --save, --quit

Error: Unrecognized arguments: --no-source, --save, --quit

Error: Unrecognized arguments: --no-source, --save, --quit
```

demarey committed:
Add --no-source flag to st command

demarey committed:
Better way to ensure bootsrap commands are found
When first commands are found through the pragma, bootstrap commands are not yer reloaded and are not found

demarey committed:
add missing option in test

demarey committed:
remove tests covered by cli tests
